flexmeasures/data/services/job_cache.py

```diff
@@ -0,0 +1,33 @@
+"""
+Logic around storing and retrieving jobs from redis cache.
+"""
+
+
+class JobCache:
+    def __init__(self, connection, queues):
+        self.connection = connection
+        self.queues = queues
+
+    def add(self, sensor_id, job_id):
+        self.connection.sadd(sensor_id, job_id)
+
+    def _get_job(self, job_id):
+        for queue in self.queues.values():
+            job = queue.fetch_job(job_id)
+            if job is not None:
+                return job
+
+    def get(self, sensor_id):
+        job_ids_to_remove = list()
+        jobs = list()
+        for job_id in self.connection.smembers(sensor_id):
+            job_id = job_id.decode("utf-8")
+            job = self._get_job(job_id)
```

Comment:
Why do this via a custom function in `JobCache`? We could just use `Job.fetch()`, see [here](https://python-rq.org/docs/jobs/#retrieving-jobs). The queue does not need to matter (and can be dropped from this class).